### ISSUE TITLE
ComputerCraft turtles (and other Forge mods) call isBlockProtected which is currently not tracked by Sponge.

### ISSUE THREAD

**Issue by inklit** [SpongePowered/SpongeCommon#812]
This leads to protection plugins such as GriefPrevention not working properly, thus causing all sorts of issues such as duplication bugs, and the ability to bypass protected areas.

The original issue was opened here
https://github.com/MinecraftPortCentral/GriefPrevention/issues/71
but the developer has came to the conclusion that it is Sponge's lack of isBlockProtected that is the problem.

Please fix this as soon as possible. Thank you.

**Comment by Zidane** [comment]
@inklit 

blood and me have been discussing a change to the API to have this work. A fix should arrive soon.

**Comment by inklit** [comment]
Is there any ETA on the fix?

Thanks.

**Comment by inklit** [comment]
http://javaprophet.com/i/2016-07-07_06-07-14.mp4

Just thought since it's been a while since this has been brought up, and no fix has arrived yet perhaps a video showing why exactly this is required might help motivate somebody to work on this.

This video was recorded on a server running the latest version of GriefPrevention (B22), and SpongeForge (1542).
Before GriefPrevention refactored their block event code (B20 and below), the Turtle would simply stop moving, and instead it would actually duplicate blocks that it attempted to break (which is equally as serious).
Now, Turtles are able to completely just bulldoze through protected areas even though this should not be possible at all.
The root cause of this problem is as the original issue states; the lack of isBlockProtected tracking.

**Comment by JavaProphet** [comment]
This is a server-breaking issue and needs to be implemented. The lack of this feature completely breaks compatibility with any Forge mod that relies on the API call and protection plugins such as GriefPrevention or FoxGaurd. This has made me consider downgrading to 1.7.10 and using Cauldron instead of Sponge because of players abuse of this problem. Please look into fixing it ASAP.

**Comment by gabizou** [comment]
Asking us to hurry up isn't going to get the issue resolved any faster. Patience is required. We already said we're working on a solution, and we're working on it. What may not be clear to the end users is that the solution is such a simple one (it's not a 5 minute fix, or even a 5 hour fix). When the fix is done and verified to work, it will be pushed and the issue closed.

**Comment by JBYoshi** [comment]
@gabizou 
> the solution is such a simple one

Did you mean it isn't simple?

**Comment by Zidane** [comment]
@JavaProphet 

That is a great way to get us to move on to other issues at hand. It isn't Sponge's fault that mods choose to say the hell with firing Forge events that exist (and would work with Sponge events automatically). We've built in a very complex tracking system to give server owners such as yourself a chance to experience mods on a large scale basis. These issues are difficult, we must approach the issue at multiple angles and ensure that all sides are covered. Your decision to come barking at us on this issue and making demands you have no right to do is neither appreciated nor warranted. If you want to be productive, take some of this demanding attitude you have and direct it at ComputerCraft. Good luck is all I'll say :).

In the mean time, we'll get back to doing what we do...fixing issues like these so YOU benefit.

**Comment by Zidane** [comment]
@inklit 

Going to other projects and saying:

```
If you would like to harass SpongeForge to implement isBlockProtected tracking, I believe that would be a good way to solve this problem.
```

Not...a good way to get us to help. Seriously now, stop taking us for granted. Be thankful we made this project and are willing to get this addressed (when it still is NOT OUR PROBLEM).

**Comment by inklit** [comment]
@Zidane 

That was said jokingly, and I linked to here because I don't know the proper procedures that Forge mods should take to modify blocks in a way that Sponge recognizes it and works well with GriefPrevention and other protection systems.

Checking for isBlockProtected being the only way I know of solving this problem, I was attempting to be as helpful as I could in supplying valid information which could assist the developer of that particular Forge mod in working properly with Sponge.

If you would like to respond to my post on that issue at that particular Forge mod and tell them how to properly fire Forge events that work with Sponge in a way that does not require isBlockProtected checks, it would certainly help that project and I could ask him if he can port that functionality to ComputerCraft.
I believe the Dan200 (developer of ComputerCraft) will probably not bother, since the project is closed source and he has already started continuing work on newer versions of ComputerCraft.
However, SquidDev's project hooks into ComputerCraft, and he also has another project that modifies the core of ComputerCraft and I believe he could remedy whatever mistake ComputerCraft has made that is causing incompatibility.

I apologize if I have offended you or anybody else on the Sponge team.
I appreciate the efforts that the Sponge team puts fourth and I am very thankful that you guys are willing to address this issue.
